upup/pkg/fi/cloudup/cloudformation/target.go

```diff
@@ -0,0 +1,173 @@
+		Project:   project,
+		outDir:    outDir,
+		resources: make(map[string]*cloudformationResource),
+	}
+}
+
+var _ fi.Target = &CloudformationTarget{}
+
+type cloudformationResource struct {
+	Type       string
+	Properties interface{}
+}
+
+// A cloudformation resource name must be alphanumeric
+func sanitizeCloudformationResourceName(name string) string {
+	name = strings.Replace(name, ".", "", -1)
+	name = strings.Replace(name, "-", "", -1)
+	name = strings.Replace(name, ":", "", -1)
+	name = strings.Replace(name, "/", "", -1)
+	return name
+}
+
+func (t *CloudformationTarget) ProcessDeletions() bool {
+	// Terraform tracks & performs deletions itself
+	return false
+}
+
+func (t *CloudformationTarget) RenderResource(resourceType string, resourceName string, e interface{}) error {
+	res := &cloudformationResource{
+		Type:       resourceType,
+		Properties: e,
+	}
+
+	name := resourceType + "::" + resourceName
+	name = sanitizeCloudformationResourceName(name)
+
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
+
+	if t.resources[name] != nil {
+		return fmt.Errorf("resource %q already exists in cloudformation", name)
+	}
+	t.resources[name] = res
+
+	return nil
+}
+
+func (t *CloudformationTarget) Find(ref *Literal) (interface{}, bool) {
+	key := ref.extractRef()
+	if key == "" {
+		glog.Warningf("Unable to extract ref from %v", ref)
+		return nil, false
+	}
+
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
+
+	r := t.resources[key]
+	if r == nil {
+		return nil, false
+	}
+	return r.Properties, true
+}
+
+func (t *CloudformationTarget) Finish(taskMap map[string]fi.Task) error {
+	//resourcesByType := make(map[string]map[string]interface{})
+	//
+	//for _, res := range t.resources {
+	//	resources := resourcesByType[res.ResourceType]
+	//	if resources == nil {
+	//		resources = make(map[string]interface{})
+	//		resourcesByType[res.ResourceType] = resources
+	//	}
+	//
+	//	tfName := tfSanitize(res.ResourceName)
+	//
+	//	if resources[tfName] != nil {
+	//		return fmt.Errorf("duplicate resource found: %s.%s", res.ResourceType, tfName)
+	//	}
+	//
+	//	resources[tfName] = res.Item
+	//}
+
+	//providersByName := make(map[string]map[string]interface{})
+	//if t.Cloud.ProviderID() == fi.CloudProviderGCE {
+	//	providerGoogle := make(map[string]interface{})
+	//	providerGoogle["project"] = t.Project
+	//	providerGoogle["region"] = t.Region
+	//	providersByName["google"] = providerGoogle
+	//} else if t.Cloud.ProviderID() == fi.CloudProviderAWS {
+	//	providerAWS := make(map[string]interface{})
+	//	providerAWS["region"] = t.Region
+	//	providersByName["aws"] = providerAWS
+	//}
+
+	data := make(map[string]interface{})
+	data["Resources"] = t.resources
+	//if len(providersByName) != 0 {
+	//	data["provider"] = providersByName
+	//}
+
+	jsonBytes, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		return fmt.Errorf("error marshalling cloudformation data to json: %v", err)
+	}
+
+	files := make(map[string][]byte)
+	files["kubernetes.json"] = jsonBytes
```

Comment:
Well _technically_...

But yes, good point.  Not sure if we should do that as a flag or not.  We could also default to YAML - it feels more readable / usable...